src/lwc/geGiftEntryFormApp/geGiftEntryFormApp.js

```diff
@@ -1,37 +1,95 @@
+const PAYMENT_STATUS__C = DI_PAYMENT_STATUS_FIELD.fieldApiName;
+const CAPTURED = 'CAPTURED';
+
+export default class GeGiftEntryFormApp extends NavigationMixin(LightningElement) {
     @api recordId;
     @api sObjectName;
 
     @track isPermissionError;
+    @track loadingText = 'Saving data import...';
+
+    get isBatchMode() {
+        return this.sObjectName &&
+            this.sObjectName === DATA_IMPORT_BATCH_OBJECT.objectApiName;
+    }
 
-    handleSubmit(event) {
+    handleBatchSubmit(event) {
         const table = this.template.querySelector('c-ge-batch-gift-entry-table');
-        const { diRecord, widgetValues } = event.detail.data;
-        let widgetString = '';
-        if(widgetValues) {
-            widgetString = JSON.stringify(widgetValues);
-        }
-        GeFormService.saveAndDryRun(
-            this.recordId, diRecord, widgetString)
-            .then(
-                dataImportModel => {
-                    Object.assign(dataImportModel.dataImportRows[0],
-                        dataImportModel.dataImportRows[0].record);
-                    table.upsertData(dataImportModel.dataImportRows[0], 'Id');
-                    table.setTotalCount(dataImportModel.totalCountOfRows);
-                    table.setTotalAmount(dataImportModel.totalAmountOfRows);
-                    event.detail.success(); //Re-enable the Save button
-                }
-            )
+        const dataImportRecord = event.detail.dataImportRecord;
+        GeFormService.saveAndDryRun(this.recordId, dataImportRecord)
+            .then(dataImportModel => {
+                Object.assign(dataImportModel.dataImportRows[0],
+                    dataImportModel.dataImportRows[0].record);
+                table.upsertData(dataImportModel.dataImportRows[0], 'Id');
+                table.setTotalCount(dataImportModel.totalCountOfRows);
+                table.setTotalAmount(dataImportModel.totalAmountOfRows);
+                event.detail.success(); //Re-enable the Save button
+            })
             .catch(error => {
                 event.detail.error(error);
             });
     }
 
+    /*******************************************************************************
+    * @description Handles a single gift entry submit. Checks to see if a payment
+    * needs to be made and processes the Data Import record.
+    *
+    * @param {object} event: Custom Event containing the Data Import record and an
+    * callback for handling and displaying errors in the form.
+    */
+    handleSingleSubmit = async (event) => {
+        try {
+            let { dataImportRecord, hasUserSelectedDonation, errorCallback } = event.detail;
+
+            const hasPaymentToProcess = dataImportRecord[PAYMENT_AUTHORIZE_TOKEN__C];
+            if (hasPaymentToProcess) {
+                this.loadingText = 'Charging card...';
+                dataImportRecord = await GeFormService.handlePaymentProcessing(
```

Comment:
Sometimes "handle" signals to me that an event is being handled after an action happened.  Might this be better named "doPurchase" or "makePurchaseCall" or something to indicate that this is initiating the payment process?

Reply:
Agree! Moved the `handlePaymentProcessing` method out of `geFormService` and importing it directly in `geGiftEntryFormApp`. Also renamed it to `makePurchaseCall` and renamed the apex method it then calls to `createTransaction`.